### PR TITLE
Skip Elastic Maps Server tests for Stack version 8.x

### LINKED ISSUE #4476: ems.TestElasticMapsServerCrossNSAssociation fails in cloud-on-k8s-e2e-tests-snapshot-versions-8x job
Job execution https://devops-ci.elastic.co/view/cloud-on-k8s/job/cloud-on-k8s-e2e-tests-snapshot-versions-8x/153/ failed:

```go
=== RUN   TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready
Retries (1h0m0s timeout):
    step.go:43: 
        	Error Trace:	utils.go:86
        	Error:      	Received unexpected error:
        	            	pod not running yet
        	Test:       	TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready
{"log.level":"error","@timestamp":"2021-05-05T03:44:54.599Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:44\ngithub.com/elastic/cloud-on-k8s/test/e2e/ems.TestElasticMapsServerCrossNSAssociation\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/ems/ems_test.go:46\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1193"}
    --- FAIL: TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready (3600.00s)
```

Because `docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT` doesn't exist yet.

```sh
> k -n e2e-venus describe pod test-cross-ns-ems-es-f7rr-ems-7fcfb8cf96-kfwxm | grep Failed
Warning  Failed  37s (x4 over 2m9s)  kubelet  Failed to pull image "docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT": rpc error: code = Unknown desc = Error response from daemon: manifest for docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT not found: manifest unknown: manifest unknown
Warning  Failed  37s (x4 over 2m9s)  kubelet  Error: ErrImagePull
Warning  Failed  24s (x6 over 2m9s)  kubelet  Error: ImagePullBackOff
```

### PR DESCRIPTION
**Pull request by thbkrkr**
This commit adds a check to skip Elastic Maps Server tests when the Stack major version is equal or greater than 8.

It made me want to factorize the code that mades the skips.

Resolves #4476.

### CODE REVIEW

File: test/e2e/ems/ems_test.go [hunk 1]
```diff
@@ -13,8 +13,7 @@ import (
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/maps"
 )
 
-// TestElasticMapsServerCrossNSAssociation tests associating Elasticsearch and Elastic Maps Server running in different namespaces.
-func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
+func maybeSkip(t *testing.T) {
```
Comment (barkbay) [inline test/e2e/ems/ems_test.go:16]: Should we manage this logic in https://github.com/elastic/cloud-on-k8s/blob/f81f51fb6e5eb39882454396b8f2f83227ffa10b/test/e2e/test/maps/builder.go#L150-L153 ?

Comment (pebrc) [reply test/e2e/ems/ems_test.go:16]: Excellent idea. I forgot about that feature of the builder! I think all of the these checks should go there. The 7.11.0 check is actually redundant.

Comment (thbkrkr) [reply test/e2e/ems/ems_test.go:16]: Yes! I also completely forgot the existence of this method.

**Review by pebrc — APPROVED** [review]

File: test/e2e/test/maps/builder.go [hunk 2]
```diff
@@ -148,6 +148,17 @@ func (b Builder) MutationReversalTestContext() test.ReversalTestContext {
 }
 
 func (b Builder) SkipTest() bool {
+	// only execute EMS tests if we have a test license to work with
+	if test.Ctx().TestLicense == "" {
+		return true
+	}
+
+	// TODO: remove once an EMS 8.x Docker image image is available
```
Comment (barkbay) [inline test/e2e/test/maps/builder.go:156]: nit: Should we also create an issue ?

Comment (thbkrkr) [reply test/e2e/test/maps/builder.go:156]: https://github.com/elastic/cloud-on-k8s/issues/4479

**Pull request merged**